### PR TITLE
updated external interfaces to add windows datatypes

### PR DESCRIPTION
**Pull request by hoopoepg**
added windows datatypes for fi_mutex_cond structure

Signed-off-by: Oblomov, Sergey <sergey.oblomov@intel.com>

### CODE REVIEW

**Comment by lanl-ompi** [comment]
bot:retest

File: include/rdma/fi_eq.h [hunk 1]
```diff
@@ -75,10 +77,18 @@ struct fid_wait {
...
 };
 
+#ifndef _WIN32
 struct fi_mutex_cond {
 	pthread_mutex_t		*mutex;
 	pthread_cond_t		*cond;
 };
+#else /* _WIN32 */
+struct fi_critsec_cond {
+	CRITICAL_SECTION	*mutex;
```
Comment (shefty) [inline include/rdma/fi_eq.h]: I think we want a different variable name here.  In true windows fashion, maybe change this to:
```c
LPCRITICAL_SECTION    critsec;
```
(I can't bring myself to use CriticalSection as the name. :)

Comment (hoopoepg) [inline include/rdma/fi_eq.h]: done...
i will create patch to fix windows build on monday (yea, this PR breaks windows build)

thank you

**Comment by shefty** [comment]
I will wait to merge this until after you add the build fix patch.  Thanks!

**Comment by hoopoepg** [comment]
additional PR: https://github.com/ofiwg/libfabric/pull/2126

File: include/rdma/fi_eq.h [hunk 2]
```diff
@@ -58,6 +60,7 @@ enum fi_wait_obj {
 	FI_WAIT_SET,
 	FI_WAIT_FD,
 	FI_WAIT_MUTEX_COND,	/* pthread mutex & cond */
+	FI_WAIT_CRITSEC_COND = FI_WAIT_MUTEX_COND,	/* critical section & cond */
```
Comment (shefty) [inline include/rdma/fi_eq.h]: Close - this needs to be its own value.  It's a distinct wait object.

Comment (hoopoepg) [inline include/rdma/fi_eq.h]: done, but it really complified code where these macro used

**Comment by goodell** [comment]
looks like this still needs the man page updates too

**Comment by shefty** [comment]
Merged manually an updated in #2141

**Pull request closed without merging**